### PR TITLE
Remove RHEL variant options

### PR DESCRIPTION
**Pull request by fellipeh**
This PR remove the option for choosing a RHEL variant, using "Server" as default.
And also inform the user in case of use this option.

### CODE REVIEW

File: convert2rhel/toolopts.py [hunk 1]
```diff
@@ -71,69 +71,68 @@ def _get_argparser():
     def _register_options(self):
         """Prescribe what command line options the tool accepts."""
         self._parser.add_option("-h", "--help", action="help", help="Show "
-                                " help message and exit.")
+                                                                    " help message and exit.")
```
Comment (sturivny) [inline convert2rhel/toolopts.py:74]: Why did you add this change here? Did you mean "    <text>" not             "<text>"?

Comment (fellipeh) [reply convert2rhel/toolopts.py:74]: It`s a vscode format issue.. fixed.

File: convert2rhel/toolopts.py [hunk 2]
```diff
@@ -30,7 +31,7 @@ def __init__(self):
         self.enablerepo = []
         self.disablerepo = []
         self.pool = None
-        self.variant = None
+        self.variant = "Server"
```
Comment (SpyTec) [inline convert2rhel/toolopts.py:34]: We should remove it completely. For all usages of variant we should just replace it with the string `server`
```suggestion
```

File: convert2rhel/toolopts.py [hunk 3]
```diff
@@ -56,84 +57,80 @@ def _get_argparser():
...
-        group.add_option("-v", "--variant", help="The RHEL variant to which"
-                         " the system will be converted. Available variants"
-                         " are: Server, Client, Workstation, ComputeNode. If"
-                         " not used, the user is asked to choose a variant.")
+                                        " the user is asked to choose from the available"
+                                        " subscriptionsincorret rhel variant. A list of the available"
+                                        " subscriptions is possible to obtain by running"
+                                        " 'subscription-manager list --available'.")
+        group.add_option("-v", "--variant", help="The RHEL variant params is not used anymore."
+                                                 " Convert2RHEL will user Server variant as default.")
```
Comment (SpyTec) [inline convert2rhel/toolopts.py:129]: Can't we simply remove this? Or do we keep this just for the logging?

If we leave it in it will still be displayed in the help page and take up space. If there's a way to remove it but still be able to detect the argument to warn the user it would be best. Should be possible with this: https://docs.python.org/2.6/library/optparse.html#optparse.OptionParser.get_option

Otherwise I think we should just remove it and simply ignore the argument and go about the conversion.
```suggestion
```

**Review by SpyTec — CHANGES_REQUESTED** [review]

File: convert2rhel/toolopts.py (hunk 2 above)
Comment (ZhukovGreen) [reply convert2rhel/toolopts.py:34]: Nit: This string "Server" is a constant, probably it has the sense to move it to the top of the module. (i.e.
```python
DEFAULT_VARIANT = "Server"
```

**Review by ZhukovGreen — COMMENTED** [review]
my 2 cents

File: convert2rhel/toolopts.py (hunk 2 above)
Comment (SpyTec) [reply convert2rhel/toolopts.py:34]: We use `toolopts.variant` in two spots I believe, I don't believe it's worth keeping the variable as we'll always use server

File: convert2rhel/toolopts.py (hunk 3 above)
Comment (fellipeh) [reply convert2rhel/toolopts.py:129]: If we remove this options, will fail when user try to use... on line 201 we check if user trying to use the params.. if we remove this options will fail there.

so, I believe it's a good approach to have this options, and warning the user as I did.. after some versions we can remove it.

Comment (SpyTec) [reply convert2rhel/toolopts.py:129]: That's why I'm asking if it's possible to use `get_option`. Getting `parsed_opts.variant` will inevitably fail if we remove this. We shouldn't have variant in help if it does nothing

Comment (fellipeh) [reply convert2rhel/toolopts.py:128]: The problem is if we use `get_options` we lost the params options.. if you try to pass  you receive something like that: 
```bash
Usage: 
  convert2rhel [-h]
  convert2rhel [-u username] [-p password | -f pswd_file] [--pool pool_id | -a] [--disablerepo repoid] [--enablerepo repoid] [--serverurl url] [--no-rpm-va] [--debug] [--restart] [-y]
  convert2rhel [--disable-submgr] [--disablerepo repoid] [--enablerepo repoid] [--no-rpm-va] [--debug] [--restart] [-y]
  convert2rhel [-k key] [-o organization] [--pool pool_id | -a] [--disablerepo repoid] [--enablerepo repoid] [--serverurl url] [--no-rpm-va] [--debug] [--restart] [-y]

convert2rhel: error: no such option: --variant
```

so we need to decide if we move forward to show a user a info message, or just remove it..

Comment (SpyTec) [reply convert2rhel/toolopts.py:128]: Found two possible solutions. One involves `argparse`, the successor to `optparse`, as it can suppress specific options to be displayed. Meaning it will still parse but not show in `--help` or man pages.

The other option, which I think we can go with now, is by checking if variant is in `sys.argv`. This means we only add variant to the parser if it's already part of the CLI arguments. Just remember to `import sys`.

```suggestion
        if '--variant' in sys.argv:
            group.add_option("-v", "--variant", help="The RHEL variant params is not used anymore."
                                                     " Convert2RHEL will user Server variant as default.")
```

File: convert2rhel/toolopts.py [hunk 4]
```diff
@@ -56,84 +56,80 @@ def _get_argparser():
...
-                         " the user is asked to choose from the available"
-                         " subscriptions. A list of the available"
-                         " subscriptions is possible to obtain by running"
-                         " 'subscription-manager list --available'.")
-        group.add_option("-v", "--variant", help="The RHEL variant to which"
-                         " the system will be converted. Available variants"
-                         " are: Server, Client, Workstation, ComputeNode. If"
-                         " not used, the user is asked to choose a variant.")
+                                        " the user is asked to choose from the available"
+                                        " subscriptionsincorret rhel variant. A list of the available"
```
Comment (SpyTec) [inline convert2rhel/toolopts.py:124]:
```suggestion
                                        " subscriptions. A list of the available"
```

File: convert2rhel/toolopts.py [hunk 5]
```diff
@@ -201,19 +199,15 @@ def _process_cli_options(self):
...
+        if '--variant' in sys.argv[1:]:
+            loggerinst.warning("Variant option is not supported anymore."
+                               "Convert2RHEL will always use Server variant from now.")
+            utils.ask_to_continue()
 
         if parsed_opts.serverurl:
             if parsed_opts.disable_submgr:
                 loggerinst.warn("Ignoring the --serverurl option. It has no effect when --disable-submgr is used.")
-            else:
+            else:
```
Comment (SpyTec) [inline convert2rhel/toolopts.py:210]:
```suggestion
            else:
```

**Review by SpyTec — APPROVED** [review]
